src/components/AppNavigation/GroupNavigationItem.vue

```diff
@@ -219,6 +245,55 @@ export default {
 			window.location.href = `mailto:?${mode}=${emails.map(encodeURIComponent).join(',')}`
 		},
 
+		/**
+		 * Rename group in store and on server
+		 */
+		renameGroup() {
+			if (this.newGroupName === '') {
+				return
+			}
+
+			this.group.contacts.forEach(async (key) => {
+				const contact = this.$store.getters.getContact(key)
+
+				if (contact === undefined) {
+					return
+				}
+
+				try {
+					await renameContactFromGroup(contact, this.group.name, this.newGroupName)
+				} catch (e) {
+					console.error('Error renaming group', e)
+				}
+			})
```

Comment:
does this need an await too?

Reply:
Would have no effect on the `forEach`. This loop could be converted to a `for (const ... of ...)` construct and then it would delay the rename commit further down properly.